src/hooks/use-chat-input.tsx

```diff
@@ -1,46 +1,23 @@
-import React, { createContext, useContext, useMemo, useRef, useCallback, ReactNode } from 'react';
+import { action, makeObservable, observable } from 'mobx';
 
-export interface ChatInputContextType {
-	getChatInput: ( siteId: string ) => string;
-	saveChatInput: ( input: string, siteId: string ) => void;
-}
-
-interface ChatInputProviderProps {
-	children: ReactNode;
-}
-
-const ChatInputContext = createContext< ChatInputContextType >( {
-	getChatInput: () => '',
-	saveChatInput: () => {
-		// NOOP
-	},
-} );
-
-export const ChatInputProvider: React.FC< ChatInputProviderProps > = ( { children } ) => {
-	const inputBySite = useRef< Record< string, string > >( {} );
-
-	const getChatInput = useCallback( ( siteId: string ) => {
-		return inputBySite.current[ siteId ] ?? '';
-	}, [] );
+class ChatInputStore {
+	inputBySite: Record< string, string > = {};
```

Comment:
Thoughts out loud - we could make agreement to make such properties private and always create methods to get/set value. But I am not sure that it's good idea, necessary to think more :)

Reply:
> we could make agreement to make such properties private and always create methods to get/set value

We could. I don't think it'd be my first choice, though, since TS already helps us enforce types effectively, and there's no need to use getters and setters to make MobX work. We can discuss this further, though.

Reply:
I have only one argument/though about it - if we have some extra method like `getChatInput` then it will be easier to find all places where it's used.
But on the other hand it's boilerplate and maybe even overengineering.